api/src/services/attachment-service.ts

```diff
@@ -116,28 +91,52 @@ export class AttachmentService extends DBService {
   }
 
   /**
-   * Finds all of the project Report attachments and Supplementary Data for the given project ID.
+   * Finds all of the project attachments and Supplementary Data for the given project ID.
    *
    * @param {number} projectId
    * @return {*}  {Promise<GetAttachmentsData[]>}
    * @memberof AttachmentService
    */
-  async getProjectReportAttachmentsWithSupplementaryData(projectId: number): Promise<GetAttachmentsData[]> {
+  async getProjectAttachmentsWithSupplementaryData(projectId: number): Promise<GetAttachmentsWithSupplementalData[]> {
     const historyPublishService = new HistoryPublishService(this.connection);
 
-    const attatchments = await this.attachmentRepository.getProjectReportAttachments(projectId);
+    const attachments = await this.attachmentRepository.getProjectAttachments(projectId);
 
-    const attachmentList: any[] = [];
+    return await Promise.all(
+      attachments.map(async (attachment: any) => {
+        const supplementaryData = await historyPublishService.getProjectAttachmentPublishRecord(
+          attachment.project_attachment_id
+        );
 
-    attatchments.forEach(async (attachment: any) => {
-      const supplementaryData = await historyPublishService.getProjectReportPublishRecord(
-        attachment.project_report_attachment_id
-      );
+        return new GetAttachmentsWithSupplementalData(attachment, supplementaryData);
+      })
+    );
+  }
 
-      attachmentList.push(new GetAttachmentsData(attachment, supplementaryData));
-    });
+  /**
+   * Finds all of the project Report attachments and Supplementary Data for the given project ID.
+   *
+   * @param {number} projectId
+   * @return {*}  {Promise<GetAttachmentsData[]>}
+   * @memberof AttachmentService
+   */
+  async getProjectReportAttachmentsWithSupplementaryData(
+    projectId: number
+  ): Promise<GetAttachmentsWithSupplementalData[]> {
+    const historyPublishService = new HistoryPublishService(this.connection);
+
+    const attachments = await this.attachmentRepository.getProjectReportAttachments(projectId);
+    console.log('attachments: ', attachments);
+
+    return await Promise.all(
```

Comment:
Can remove the `await` and just return the Promise

Reply:
done